src/main/java/com/github/nishgpt/chainexecutor/models/execution/StageExecutor.java

```diff
@@ -48,6 +48,13 @@
    */
   void validateStatus(T context);
 
+  /**
+   * Checks if the stage can be skipped
+   *
+   * @return
+   */
+  T skipIfApplicable(T context);
```

Comment:
This can be provided as default method also, where default impl would be a no-op 
```
default T skipIfApplicable(T context) {
return context;
}
```